module/VuFindApi/src/VuFindApi/Controller/McpController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * Controller for Model Context Protocol (MCP)
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFindApi\Controller;
+
+use Laminas\Diactoros\ServerRequestFactory;
+use Laminas\Psr7Bridge\Psr7Response;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Mcp\Server;
+use Mcp\Server\Transport\StreamableHttpTransport;
+use Psr\Http\Message\ResponseInterface;
+use VuFind\Controller\AbstractBase;
+
+/**
+ * Controller for Model Context Protocol (MCP)
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class McpController extends AbstractBase
+{
+    // protected $accessPermission = 'access.mcp';
```

Comment:
Should probably add some real permissions.  Maybe specific to individual capabilities, maybe just one general permission.

Reply:
One permission created for now.